### DISCUSSION TITLE
How to elimilate the file path before each line of the search results when redirecting rg to a file or pager(less)

### DISCUSSION THREAD

**Discussion by AstroShen** [BurntSushi/ripgrep#2110] [Q&A] [answered]
When i use command like this:
**rg 'pattern'  > somefile.log**
Every line in the somefile.log is started with a matching file path that this line belongs to. This feature pollute the search results when i want to store the searching results.
I have tried a couple of options such as --no-heading, --no-filename, which works only when i dont redirect the search results to a file.
Please help me.

**Comment by BurntSushi** [comment, marked as answer]
Works fine for me:

```
$ echo test > foo

$ echo test > bar

$ rg test
foo
1:test

bar
1:test

$ rg test --no-filename
1:test

1:test

$ rg test > /tmp/rg-with-file.log

$ rg test --no-filename > /tmp/rg-without-file.log

$ cat /tmp/rg-with-file.log
foo:test
bar:test

$ cat /tmp/rg-without-file.log
test
test
```

You'll need to provide a more specific reproduction. Try the above commands, for example.
